app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt

```diff
@@ -190,8 +195,17 @@ class MainActivity :
         val view = binding.root
         setContentView(view)
 
-        if (BuildConfig.END_OF_YEAR_ENABLED && settings.getEndOfYearShowBadge2022()) {
-            binding.bottomNavigation.getOrCreateBadge(VR.id.navigation_profile)
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.isEndOfYearStoriesEligible().collect { isEligible ->
+                    if (isEligible) {
+                        setupEndOfYearLaunchBottomSheet()
+                        if (settings.getEndOfYearShowBadge2022()) {
+                            binding.bottomNavigation.getOrCreateBadge(VR.id.navigation_profile)
+                        }
+                    }
+                }
+            }
```

Comment:
This could be refactored to use `.flowWithLifeycle()` and `.launchIn()` extensions to avoid nesting once functions exposing `Flow` in `StoriesDataSource` are no longer suspending.

Reply:
Nice tip, applied in ebe4f45